Documentation/proposal-runconfig_inside_osv.md

```diff
@@ -0,0 +1,40 @@
+# Proposal for meta/run.yaml content being present in the composed image
+
+Currently, Capstan runtime support makes use of `$CWD/meta/run.yaml` for composing and running an OSv image. Although this is fine for the composing, it gets tricky for running because `capstan run <IMAGE>` generally does not require user to be inside package root directory.
+
+I propose the following solution:
+
+### Composing image
+File meta/run.yaml is present at package root directory (same as until now)
+
+When user runs `capstan package compose` the content of meta/run.yaml file is processed and new files are generated inside the OSv image folder `/osv-runtime`:
+
+Capstan should create a separate file for each named configuration e.g.
+```
+/osv-runtime
+   |--burek
+   |--pizza   <--------\
+   |                    |
+   |--default-runtime  /
+```
+Each runtime file should contain only one line representing the run command that is to be run inside OSv:
+```
+# content of file /osv-runtime/burek:
+node burek.js
+```
+
+When user runs `capstan package compose` the boot command is set so that the agent is run. A capstan package must be implemented that provides agent funcionality.
+
+#### OSv Agent
+Agent reads *cloud-init* file. Based on cloud-init data it spawns a thread with appropriate run command.
```

Comment:
I was always thinking about cloud-init as a file that is passed to the instance - so I'm not sure what second disk image are we talking about here. Let me read some docs.

BTW: How come that OSv currently supports cloud-init, but does not support attaching volumes?

#### Assuming that it is difficult to get cloud-init working locally
When running with capstan, we could somehow pass name of the configuration thru environment variable. The Agent would then need to be able to pick name from either environment either cloud-init. But that brings us back to the necessity of `capstan run` that we would like to avoid...

Ideas?